csharp-advanced-topics/FuncAndActionDelegates/Tests/FuncAndActionDelegateUnitTest.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using Xunit;
+using FuncAndActionDelegate;
+
+namespace Tests
+{
```

Comment:
I had tested Func class only in this section (all the methods of Func class)
I had deliberately left the Action class because all of the methods were void (Action delegates don't return anything). I could've used actionDelegate.GetInvocationList() to check if the delegate was getting called but this would have unnecessarily increased the complexity, so I didn't use this...

If we are expecting 100% code coverage, I'll include all the test cases, going forward.